### DISCUSSION TITLE
add py.typed so that your type annotations can inform application codebases

### DISCUSSION THREAD

**Discussion by petergaultney** [zauberzeug/nicegui#2128] [Ideas / Feature Requests]
If you [add](https://blog.whtsky.me/tech/2021/dont-forget-py.typed-for-your-typed-python-package/) a `py.typed` file to your source code root, then applications that people build will be able to have `mypy` and friends check that they're using your APIs basically correctly. This is pretty easy to do and since you're already doing a great job of putting type annotations in your code, it's likely this would be an improvement for lots of users!

**Comment by falkoschindler** [comment]
Sounds great, @petergaultney! We haven't been aware of `py.typed`, but we love to improve support for type annotations wherever we can. Would you like to create a pull request? 🙂

**Comment by petergaultney** [comment]
happy to!

https://github.com/zauberzeug/nicegui/pull/2142

**Comment by petergaultney** [comment]
closed by https://github.com/zauberzeug/nicegui/pull/2142 !
